test/io/write_gate_level_layout.cpp

```diff
@@ -0,0 +1,150 @@
+    check_parsing_equiv<Lyt>(blueprints::inverter_network<technology_network>());
+    check_parsing_equiv<Lyt>(blueprints::clpl<technology_network>());
+    check_parsing_equiv<Lyt>(blueprints::one_to_five_path_difference_network<technology_network>());
+    check_parsing_equiv<Lyt>(blueprints::nand_xnor_network<technology_network>());
+    check_parsing_equiv_layout(blueprints::straight_wire_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::three_wire_paths_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::xor_maj_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::single_input_tautology_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::tautology_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::and_or_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::and_or_gate_layout<cart_odd_row_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::and_or_gate_layout<cart_even_row_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::and_or_gate_layout<cart_odd_col_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::and_or_gate_layout<cart_even_col_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::and_or_gate_layout<hex_odd_row_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::and_or_gate_layout<hex_even_row_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::and_or_gate_layout<hex_odd_col_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::and_or_gate_layout<hex_even_col_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::and_not_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::or_not_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::use_and_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::res_maj_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::res_tautology_gate_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::open_tautology_gate_layout<hex_even_row_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::crossing_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::fanout_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::unbalanced_and_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::shifted_cart_and_or_inv_gate_layout<cart_odd_col_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::row_clocked_and_xor_gate_layout<cart_even_row_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::optimization_layout<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::optimization_layout_corner_case_outputs_1<cart_gate_clk_lyt>());
+    check_parsing_equiv_layout(blueprints::optimization_layout_corner_case_outputs_2<cart_gate_clk_lyt>());
```

Comment:
It's a bit odd that more than half of these calls don't use the `Lyt` parameter, but the others do.